docs/contributing/code_guide.rst

```diff
@@ -23,47 +23,96 @@ an idea at a community meeting.
 Coding Style
 ============
 
-PlasmaPy Code Style Guide, codified
------------------------------------
+* Use formatted string literals (f-strings) instead of legacy formatting
+  for strings.
 
-* PlasmaPy follows the `PEP8 Style Guide for Python Code
-  <https://peps.python.org/pep-0008>`__.  This style choice
-  helps ensure that the code will be consistent and readable.
+  >>> package_name = "PlasmaPy"
+  >>> print(f"The name of the package is {package_name}.")
+  The name of the package is PlasmaPy.
+  >>> print(f"{package_name=}")
+  package_name='PlasmaPy'
+  >>> print(f"{package_name!r}")  # shortcut for f"{repr(package_name)}"
+  'PlasmaPy'
 
-  * Line lengths should be chosen to maximize the readability and
-    elegance of the code.  The maximum line length for Python code in
-    PlasmaPy is 88 characters.
+* Do not use :term:`mutable` objects as default values in the function
+  or method declaration. This can lead to unexpected behavior.
```

Comment:
This is technically not a stylistic choice, but a bug-avoidance one. No idea where else to put it though. Could be skipped and caught in reviews, I guess. Could be kept as is.

Reply:
Hm...perhaps we could add a section in the coding guide on Best Practices and put it there?  

A side note is that we recently added the [flake8-mutable](https://github.com/ebeweber/flake8-mutable) extension which checks for mutable default arguments...which is really helpful!  I still think this would be worth including in the coding guide, though.

Reply:
Guess I managed to miss that one :) I guess it's fine in here. If someone adds a mutable def arg, they're going to get told about it anyway by CI :)